### ISSUE TITLE
Not possible to share current folder (need to display share icon in the breadcrumbs)

### ISSUE THREAD

**Issue by jancborchardt** [owncloud/core#4333]
1. It will be easy to see what is shared
2. If we add sharing functionality from there, it will be easy to share the folder you’re currently in

Currently there is no way to share a folder from inside it. You have to move outside of it, find it in the list again, and then share it from there.

Solution proposal: Add a »Share« button (only with icon) in the breadcrumb next to the name of the current folder. When that is clicked, open the share modal. That button should also reflect the state of the folder just like the action button, for example showing a world when the folder is shared by public link.

@MTGap what do you think? Also cc @owncloud/designers

**Comment by DeepDiver1975** [comment]
@jancborchardt close or 8.1? THX

**Comment by MorrisJobke** [comment]
8.1

**Comment by jancborchardt** [comment]
Yeah, no close since it’s not done and it would be a very good addition to make sharing easier.

**Comment by PeterRay** [comment]
Hi all, I've tryed to fix. I add an icon and also relevant event handler but I stopped because javascript doesn't return the event handler of a button_mouse_click. (I need this to retrive the common function (for sure already developed in controller) that set a folder as shared- I expect to find something like ShareFunction(path,....) ).  Sorry but I have no more time to spend to follow javascript lacks. If somebody need code, let me know. thanks. greetings for owncloud design itself

**Comment by jancborchardt** [comment]
@PeterRay do you have the code on Github? cc @MorrisJobke @PVince81 about the JS issues maybe you can help?

**Comment by jancborchardt** [comment]
@PeterRay even if the code is not ready, you can open a pull request. Then people can help out. :)

**Comment by nickvergessen** [comment]
Removed the shortlist label, because the milestone was changed

**Comment by cmonteroluque** [comment]
Not sure this has enough momentum to be in 8.2?

**Comment by MorrisJobke** [comment]
> Not sure this has enough momentum to be in 8.2?

I don't think so.

**Comment by jancborchardt** [comment]
Unfortunately – setting to 9.0 then.

@oparoz maybe this is good in conjunction with moving the share function in the Gallery app as well, in case you want to look into that.

**Comment by oparoz** [comment]
That will improve the consistency of the UI across apps :)

**Comment by jancborchardt** [comment]
Longstanding issue, but I’d judge as small investment with big impact. cc @DeepDiver1975 @schiesbn for sharing.

I’d really appreciate if we can get this done for 9.0

**Comment by schiesbn** [comment]
Now that sharing was moved to the sidebar, wouldn't it make sense to display in the sidebar some kind of folder state if no file is selected? I think this was suggested by @oparoz in a different issue. This way you could easily share the parent folder if no file is selected directly from the sidebar.

Another idea (which could be combined with the first one or implemented independently):

If the user moves back to the parent folder, via breadcrumbs, we could automatically select the folder where he comes from. This way he could jump back to the parent, the folder he come from is automatically selected and he could directly create the share in the sidebar.

Not sure if one or both of  this ideas are easy to implement? @PVince81 can probably say more and maybe even implement it if we agree on it :wink:

**Comment by nickvergessen** [comment]
I guess when we want to have activities and comments (aka timeline) for folders as well, it would also make sense to make them accessible inside the folder?

**Comment by jancborchardt** [comment]
@schiesbn @nickvergessen that is a slightly different discussion though. What I’m simply asking for here is a share icon next to the current folder in the breadcrumb which will open the sidebar for the current folder.

Showing the sidebar always will be unnecessary clutter in most cases, or something which we can discuss separately.

(Also @schiesbn – people using ownCloud are not only male, so I’d suggest using »they« instead of »he« as a more inclusive personal pronoun.)

**Comment by schiesbn** [comment]
> that is a slightly different discussion though. What I’m simply asking for here is a share icon next to the current folder in the breadcrumb which will open the sidebar for the current folder.

Can't imagine that this will look good, but you are the designer ;)

Also what happens if you click on the icon? We no longer have a share dialog. So it would open the sidebar or change the content of the sidebar if it is already open. To what? Probably to some kind of folder state. That's why I suggested the folder state solution. We need to open the sidebar anyway for sharing but I think we can do this in a more elegant way than a clumsy share icon next to the breadcrumbs.

Just my 2ct.

**Comment by jancborchardt** [comment]
> Also what happens if you click on the icon? We no longer have a share dialog. So it would open the sidebar or change the content of the sidebar if it is already open. To what? Probably to some kind of folder state.

Exactly.

How is having a dedicated way to open the share sidebar for the current folder a more clumsy way than just automatically putting it in people’s faces? Because how can you change to the folder sharing view when you have the sidebar open with details of a file?

**Comment by schiesbn** [comment]
> How is having a dedicated way to open the share sidebar for the current folder a more clumsy way than just automatically putting it in people’s faces? Because how can you change to the folder sharing view when you have the sidebar open with details of a file?

Could be two possibilities, as described above:

1. click on a empty area to switch to unselect the file and view the folder state instead
2. click on the folder name at the breadcrumbs. This will move you to the parent folder, auto select the folder you come from and you can go right to the sidebar to create the share.

**Comment by jancborchardt** [comment]
> click on a empty area to unselect the file and view the folder state instead

This is not discoverable. ;)

> click on the folder name at the breadcrumbs

This is basically what I’m proposing. But it’s important to show the share icon there so it’s clear that it’s actionable. Also, this serves as an indicator if the current folder is shared.

**Comment by PVince81** [comment]
@jancborchardt can you clarify the following: the ticket subject says "show share icon in breadcrumb". When I read this I picture the share icon being directly inside the breadcrumb of every parent folder (or only the last one ?). Should the share icon/button be more on the right, like the "+" button instead ?

I think it's fine to allow showing the current folder info in the sidebar as long as it can be activated without relying on the share button in the controls bar as it could be missing if sharing is disabled.
So basically the user could click both the share button and the folder name ?

**Comment by MorrisJobke** [comment]
> So basically the user could click both the share button and the folder name ?

But then how to navigate upwards in the file tree?

**Comment by jancborchardt** [comment]
It would only show for the current folder, which currently is not clickable anyway. In the future we could think about adding it for folders above that as well – in that case of course only the share icon would be clickable.

**Comment by PVince81** [comment]
It might be possible to create a quick prototype PR with the icon to get a better feel of the interaction.
However, the information of the current folder is not yet available (need the Webdav PR first: #16902 to get access), but it might not be needed for testing :smile:

**Comment by jancborchardt** [comment]
@PVince81 any status here? Otherwise we should plan it for 9.1 cc @MTRichards

**Comment by ChristophWurst** [comment]
@jancborchardt what should that look like? Where should that icon be placed?

Something like that?
![bildschirmfoto von 2016-04-13 15-03-43](https://cloud.githubusercontent.com/assets/1374172/14494017/e86a7868-0188-11e6-8071-bc60bd6e5a7f.png)

**Comment by nickvergessen** [comment]
I think it would be also cool to have the ... menu available, so you can access tags, comments and other info as well

**Comment by jancborchardt** [comment]
@ChristophWurst exactly! These are my mockups:
![share current folder](https://cloud.githubusercontent.com/assets/925062/14526361/07191994-0243-11e6-921e-dc0e8d118bda.png)
![share current folder - shared link](https://cloud.githubusercontent.com/assets/925062/14526360/07188eca-0243-11e6-8a93-cb6b2a450a36.png)

@nickvergessen we’re not going to do the full 3-dot menu _at the moment_ for two reasons: 1) it’s too much in there. 2) the share icon also serves as an indicator.
Maybe later in some shape or form.

**Comment by oparoz** [comment]
* People will never guess this is a button until they inadvertently click on it while trying to access a parent folder
* The difference between the button's default state of shareable and a "shared" folder is going to be just as confusing as it is on the row
* There will be another location for the 3-dots action menu for the current folder, bringing more confusion to the UI

### Solution

* Make the share icon on the row a pure indicator. If it's not there, the folder is not shared. There is no need to add the icon to the breadcrumbs any more. You land in a folder and know in an instant that it's shared
* Use the 3-dots icon for all actions. There is plenty of room next to the [+] or the [+] can be converted to a 3-dots. There is nothing special about adding new stuff to a folder, those are just actions

**Comment by nickvergessen** [comment]
Are we going to open the sidebar on clicking the share icon?

**Comment by jancborchardt** [comment]
> Are we going to open the sidebar on clicking the share icon?

This is the entire point of this addition. A quick way to share the current folder.

**Comment by jancborchardt** [comment]
> People will never guess this is a button until they inadvertently click on it while trying to access a parent folder

Well, if they do not know then it’s an indicator. As you also suggest. If they _do_ know, it’s a shortcut to the sharing function.

**Comment by PVince81** [comment]
We're past feature freeze, moving this UX enhancement to 9.2

CC @MTRichards @cmonteroluque

**Comment by PVince81** [comment]
@pmaier1 as we mentioned this when discussing sharing.

Would move to backlog as there is no time to finish the discussion and implement this.

**Comment by pmaier1** [comment]
@felixheidecke 
Setting this to 'planned'. Maybe it's an easy low-hanging fruit.

**Comment by PVince81** [comment]
moving to triage for rescheduling

it's not exactly a low hanging fruit as it involves a bit more diggingin the messy code to get at the data of the current folder.

Estimate: 2-4md depending on UI design for the icon (putting icons up there is tricky due to screen sizes, etc)
